internal/alert/alert.go

```diff
@@ -17,26 +18,45 @@ type Rule struct {
 	Alert        *v1.Alert
 }
 
-func FlattenRules(groups []v1.RuleGroup) []Rule {
+func FlattenRules(groups []v1.RuleGroup, wantedGroups []string) []Rule {
```

Comment:
@gunzl1ng3r Here's some inspiration: 

```go
// Flattens a list of RuleGroup containing a list of Rules into
// a list of internal Alertingrules.
var l int

// Set initial capacity to reduce memory allocations.
for _, grp := range groups {
        l += len(grp.Rules)
}

// If we have wanted groups the capacity is at max
// the length of the wanted groups
if len(wantedGroups) > 0 {
        l = len(wantedGroups)
}

rules := make([]Rule, 0, l)

var r Rule

for _, grp := range groups {
        // If we have wanted groups we only care about
        // these groups, thus we can skip if not in wanted groups.
        if len(wantedGroups) > 0 {
                if !slices.Contains(wantedGroups, grp.Name) {
                        continue
                }
        }

        for _, rl := range grp.Rules {
                // For now we only care about AlertingRules,
                // since RecodingRules can simply be queried.
                if _, ok := rl.(v1.AlertingRule); ok {
                        r.AlertingRule = rl.(v1.AlertingRule)
                        rules = append(rules, r)
                }
        }
}

return rules
```

Reply:
I know it's just an example, but this wouldn't work, would it? As is read it `l` should contain the total number of rules found in all groups. With this example it would contain the number of `wantedGroups`.

I pushed an update, that should gets rid of the `else` constructs.